cc/engine.hs

```diff
@@ -0,0 +1,83 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Main where
+
+import Data.Aeson
+import Data.List (isSuffixOf)
+import Data.Maybe (fromMaybe)
+import System.Process (callProcess)
+import System.IO (hPutStrLn, stderr)
+
+import qualified Data.ByteString.Lazy.Char8 as C8
+
+data Config = Config
+    { cIncludePaths :: [FilePath]
+    , cExtraFlags :: [String]
+    }
```

Comment:
Is this a config format we are defining? Any reason not to reuse something that already exists? Or is this a code climate format?

Reply:
This is (our materialization of) the defined `/config.json` format we operate with as a CC Engine. I tried to explain how it works a little bit in the comment down with the `ToJSON` instance.